nomad/leader_test.go

```diff
@@ -903,6 +904,66 @@ func TestLeader_UpgradeRaftVersion(t *testing.T) {
 	}
 }
 
+func TestLeader_Reelection(t *testing.T) {
```

Comment:
Still need to fix up this test to be a table driven test that tests both protocol 2 and 3 leader elections

Reply:
👍 I believe this was your intent, but just to be explicit, lets do it as part of this PR